internal/cmd/controller/reconciler/gitrepo_controller.go

```diff
@@ -329,6 +351,38 @@ func purgeImageScans(ctx context.Context, c client.Client, gitrepo types.Namespa
 	return nil
 }
 
+func purgeNamespace(ctx context.Context, c client.Client, deleteNamespace bool, ns string) error {
+	if !deleteNamespace {
+		return nil
+	}
+
+	if ns == "" {
+		return nil
+	}
+
+	// Ignore default namespaces
+	defaultNamespaces := []string{"fleet-local", "cattle-fleet-system", "default"}
```

Comment:
We may want to add a couple of namespaces to this set as per [this page](https://fleet.rancher.io/namespaces), eg.:
* `fleet-default`
* `cattle-fleet-clusters-system`